src/OpenFeature.Contrib.Providers.Flagd/JsonEvaluator.cs

```diff
@@ -0,0 +1,223 @@
+        public ResolutionDetails<double> ResolveDoubleValue(string flagKey, double defaultValue, EvaluationContext context = null)
+        {
+            return ResolveValue(flagKey, defaultValue, context);
+        }
+
+        public ResolutionDetails<Value> ResolveStructureValue(string flagKey, Value defaultValue, EvaluationContext context = null)
+        {
+            return ResolveValue(flagKey, defaultValue, context);
+        }
+
+        private ResolutionDetails<T> ResolveValue<T>(string flagKey, T defaultValue, EvaluationContext context = null)
+        {
+            // check if we find the flag key
+            var reason = Reason.Default;
+            if (_flags.TryGetValue(flagKey, out var flagConfiguration))
+            {
+                reason = Reason.Static;
+                var variant = flagConfiguration.DefaultVariant;
+                if (flagConfiguration.Targeting != null && !String.IsNullOrEmpty(flagConfiguration.Targeting.ToString()) && flagConfiguration.Targeting.ToString() != "{}")
+                {
+                    reason = Reason.TargetingMatch;
+                    var targetingString = flagConfiguration.Targeting.ToString();
+                    // Parse json into hierarchical structure
+                    var rule = JObject.Parse(targetingString);
+                    // the JsonLogic evaluator will return the variant for the value
+
+                    // convert the EvaluationContext object into something the JsonLogic evaluator can work with
+                    dynamic contextObj = (object)ConvertToDynamicObject(context.AsDictionary());
+
+                    variant = (string)_evaluator.Apply(rule, contextObj);
+                }
+
+
+                // using the returned variant, go through the available variants and take the correct value if it exists
+                if (variant != null && flagConfiguration.Variants.TryGetValue(variant, out var foundVariantValue))
+                {
+                    if (foundVariantValue is Int64)
+                    {
+                        checked
+                        {
+                            foundVariantValue = Convert.ToInt32(foundVariantValue);
+                        }
+                    }
+                    else if (foundVariantValue is JObject value)
+                    {
+                        foundVariantValue = ConvertJObjectToOpenFeatureValue(value);
+                    }
+                    return new ResolutionDetails<T>(
+                        flagKey: flagKey,
+                        value: (T)foundVariantValue,
+                        reason: reason,
+                        variant: variant
+                        );
+                }
+                reason = Reason.Default;
+            }
+            return new ResolutionDetails<T>(
+                flagKey: flagKey,
+                value: defaultValue,
+                reason: reason,
+                errorType: ErrorType.FlagNotFound
+                );
```

Comment:
Don't return error codes, throw `FlagNotFound` instead (this causes the error handlers to run).

I know this is confusing. I'd like to release a new SDK version at some point that prevents providers from returning errors like this (forcing throws).